src/librustdoc/clean/mod.rs

```diff
@@ -1745,12 +1743,7 @@ impl<'tcx> Clean<Type> for Ty<'tcx> {
 
 impl<'tcx> Clean<Constant> for ty::Const<'tcx> {
     fn clean(&self, cx: &mut DocContext<'_>) -> Constant {
-        Constant {
-            type_: self.ty.clean(cx),
-            expr: format!("{}", self),
-            value: None,
-            is_literal: false,
-        }
+        Constant::TyConst { type_: self.ty.clean(cx), expr: format!("{}", self) }
```

Comment:
Rather than stringifying `self`, could you store it in the `Constant` and only stringify on demand? Or does that make the lifetimes painful?

Reply:
It's worth giving it a try.

Reply:
So, after taking a quick look, I propose this instead: I'll add a FIXME so we did it in another PR because this will otherwise just grow the diff once again, and it's already huge.